DOCS-2787: QA configure simple smart machines (#3348)
viamrobotics/docs: docs/how-tos/configure.md

Reviewer: Needs an "In this page" section, even if it's only one line long like https://docs.viam.com/how-tos/upload-module/

Reviewer 2: I think we should also have a prerequisites section even if it says none or just "A device that can run `viam-server` or `viam-micro-server`" and links to the capabilities section of the install guide

Author: done

---

> title: "Build and configure a smart machine"

Reviewer: [opt] 
```suggestion
title: "Configure a smart machine"
```

Reviewer 2: Maybe create and configure. I think only we think of the configuration aspects as how you build a representation of a machine within Viam. 
```suggestion
title: "Create and configure a smart machine"
```

linkTitle: "Configure a smart machine"
weight: 10
type: "docs"

> description: "Build a machine in a few steps using Viam's modular system of components and services without writing much or any code."

Reviewer 2: ```suggestion
description: "Create a machine in a few steps using Viam's modular system of components and services without writing much or any code."
```

> 

Reviewer 2: Can we change the first bit to:

```
All of the software that runs your machine on a computer is packaged into a binary called `viam-server`.
If you are using a microcontroller, use `viam-micro-server` instead.
Install it on the computer controlling your smart machine by following the {{< glossary_tooltip term_id="setup" text="setup instructions" >}} in the [Viam app](https://app.viam.com/).
```

> If you are using a microcontroller instead of a 64-bit computer, you can install a [lightweight version of `viam-server`](/installation/#install-viam-micro-server).

Reviewer 2: ```suggestion
```

You can install `viam-server` on your personal computer, or on a single-board computer (SBC).

> After following the setup instructions and starting `viam-server` on your machine, your machine should appear as **Live** on the Viam app.

Reviewer 2: ```suggestion
After following the setup instructions, your machine should connect and appear as **Live** on the Viam app.
```

{{% /tablestep %}}
{{% tablestep link="/configure/" %}}
**3. Navigate to the CONFIGURE tab of your machine**

Machines can be small and simple or very complex but they are all configured on the **CONFIGURE** tab of the [Viam app](https://app.viam.com/).
A machine can be a single-board computer with a single sensor or LED wired to it, or a machine can consist of multiple computers with many physical components connected, acting as one unit.

<div>

> {{< imgproc src="/viam/machine-components.png" alt="Machine components" resize="600x" class="aligncenter" >}}

Reviewer: We've gotten a lot of feedback on this diagram, some of which is that people are confused by it. What do you think of replacing this instance of it with a screenshot of a config tab with some basic components configured (maybe board, motors, camera?)

Author: done

For each component that makes up your machine:

1. Find an appropriate model for your hardware.
   You can find the available models on the [component pages](/components/).
   For example, you can scroll through available sensor models on the [sensor page](/components/sensor/#available-models).
2. Add a suitable model to your machine on the **CONFIGURE** page:

   - Click the + icon next to your machine part in the left-hand menu and select Component.

>    - [Choose any component type](/configure/#components) (example: `camera`) and model (example: `webcam`).

Reviewer 2: Don't think the link is needed - we already have the link a few lines above
```suggestion
   - Choose any component type (example: `camera`) and model (example: `webcam`).
```

> 3. Click on the **Test** area of the configuration panel to test controlling your component.

Reviewer: A lil awk I think. How about this?
```suggestion
3. Click the **TEST** area of the configuration panel to test your component, for example to view a camera feed or turn a motor.
```

> 4. If any problems occur check the [logs](/cloud/machines/#logs) or review or roll back the [configuration history](/cloud/machines/#configure).

Reviewer 2: ```suggestion
4. If any problems occur check the [**LOGS** tab](/cloud/machines/#logs). You can also review the [configuration history](/cloud/machines/#configure) and roll back changes if needed.
```

> If a component you want to use for your project is not natively supported, you can [build your own modular resource](/how-tos/create-module/).

Reviewer 2: ```
You need to [_configure_](/configure/) your machine so that `viam-server` can interact with its hardware.
Use the configuration builder tool in the Viam app to create a file that describes what hardware you are using and how it is connected.
For example, if you have a DC motor, follow the [corresponding configuration instructions](/components/motor/gpio/) to tell the software which pins it is connected to.
```

The following lines don't make sense anymore. They presumably should be with step 2. Meaning there should be another step before testing that says

```
1. When you add a component model, it will create a panel in the configuration builder tool.
    Fill in any required attributes, following the documentation for the specific model. 
```

Author: sorry, was slightly confused but I think I implemented what you want here! lmk if it looks good or if I need to make adjustments

For each component that makes up your machine:

1. Find an appropriate model for your hardware.
   You can find the available models on the [component pages](/components/).
   For example, you can scroll through available sensor models on the [sensor page](/components/sensor/#available-models).
2. You need to [_configure_](/configure/) your machine so that `viam-server` can interact with its hardware.
   Use the configuration builder tool in the Viam app to create a file that describes what hardware you are using and how it is connected.
   For example, if you have a DC motor, follow the [corresponding configuration instructions](/components/motor/gpio/) to tell the software which pins it is connected to.
3. Add a suitable model to your machine on the **CONFIGURE** page:

   - Click the + icon next to your machine part in the left-hand menu and select Component.

>    - Choose any component type (example: `camera`) and model (example: `webcam`).

Reviewer 2: ```suggestion
   - Choose any component type (example: `camera`) and model (example: `webcam`). If a component you want to use for your project is not natively supported, you can [build your own modular resource](/how-tos/create-module/).
```

4. When you add a component model, it will create a panel in the configuration builder tool. Fill in any required attributes, following the documentation for the specific model.
5. Click the **TEST** area of the configuration panel to test your component, for example to view a camera feed or turn a motor.
6. If any problems occur check the [**LOGS** tab](/cloud/machines/#logs). You can also review the [configuration history](/cloud/machines/#configure) and roll back changes if needed.

> If a component you want to use for your project is not natively supported, you can [build your own modular resource](/how-tos/create-module/).

Reviewer 2: Part way there.
This also doesn't make sense here. It should be with when you are looking for suitable models I think
```suggestion
```

> If a component you want to use for your project is not natively supported, you can [build your own modular resource](/how-tos/create-module/).

Reviewer 2: This can probably also go further up
```suggestion
```

<!-- markdownlint-disable MD036 -->

**5. Control your components**

> When you configure a component, a remote control panel is generated for it in the **CONTROL** tab of the Viam app.

Reviewer: I think with the test section in existence, it makes sense to mention it here. Something like, 
```suggestion
When you configured each component, you saw the **TEST** panel on its configuration panel.
You can also access the control interfaces for all your components in one place from the **CONTROL** tab.
```

> With the panels, you can drive motors at different speeds, view your camera feeds, see sensor readings, and generally test the basic functionality of your machine before you've even written any code.

Reviewer 2: Inside services the sentence introducing the bullets or the bullets need to change. They require a "which" or added colons. For example:

```
**Data Management**: Capture and sync data from one or more machines, and use that data for machine learning and beyond.
```

Services are built-in Viam software packages that add high-level functionality to your smart machine such as:

> **Data Management**: Capture and sync data from one or more machines, and use that data for machine learning and beyond.
> 

Reviewer 2: ```suggestion
- **Data Management**: Capture and sync data from one or more machines, and use that data for machine learning and beyond.
```

Author: ahh sorry

> {{% card link="/how-tos/develop-app/" %}}

Reviewer 2: I think we should add fragements here too and then change the next steps intro
```suggestion
{{% card link="/how-tos/develop-app/" %}}
{{% card link="/how-tos/one-to-many/" %}}
```

Author: sorry, what do you mean by adding fragments here too? What is the motivation behind directing them to fragments? that would help me understand how to incorporate

Author: oh ok i might get it just directing them to one to many how to? lmk what you think

Reviewer 2: Yes that how-to is essentially the guide to fragments.

Please spend a few minutes thinking about the context of what I am saying before asking me to explain it further. You have enough product knowledge to know why fragments should be mentioned to someone who has just configured a machine.

Author: i was a bit confused bc that doesn't immediately make sense to me, this is like the very first step of building a machine and then making a fragment to configure a lot of machines is a lot more effort and commitment to using 
Viam. That's why I was asking for more information.

Author: your suggestion above makes a lot of sense 👍

## Next steps

For more information, see the [configuration documentation](/configure/).

> Once you have configured your machine, continue to develop an application or learn about _{{< glossary_tooltip term_id="fragment" text="fragments" >}}_ and configure multiple similar machines:

Reviewer 2: ```suggestion
Once you have configured your machine, continue to develop an application or learn about how you can use _{{< glossary_tooltip term_id="fragment" text="fragments" >}}_ to configure multiple similar machines in one go:
```
